langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/vfs/Database.scala

```diff
@@ -20,16 +21,43 @@ final case class Database(entries: List[Entry]) {
   def toSchema: s.Database = {
     val sentries = entries.flatMap { ventry =>
       val sdb = s.Database.parseFrom(ventry.bytes)
-      sdb.documents
+      // returns true if this document contains only messages and nothing else.
+      // deprecation messages are reported in refchecks and get persisted
+      // as standalone documents that need to be merged with their typer-phase
+      // document during loading. It seems there's no way to merge the documents
+      // during compilation without introducing a lot of memory pressure.
+      def isOnlyMessages(sdocument: s.Document): Boolean =
```

Comment:
Why do we have to duplicate this logic wrt `sdocumentsMerged`?

Reply:
I've consolidated the logic under the extension method `s.Database.mergeMessageOnlyDocuments: s.Database`.